numpy/testing/nose_tools/utils.py

```diff
@@ -675,14 +675,20 @@ def assert_approx_equal(actual,desired,significant=7,err_msg='',verbose=True):
     if np.abs(sc_desired - sc_actual) >= np.power(10., -(significant-1)):
         raise AssertionError(msg)
 
+def asarray_orobj(x):
+    try:
+        return array(x, copy=False, subok=True)
+    except:
+        return array(x, copy=False, dtype='O', subok=True)
+    
```

Comment:
Not sure if this is the right way to fix the problem below, that `assert_array_equal` sometimes is given lists which need to be converted to object arrays. I did it this way just to quickly get tests to run.

Reply:
Should never use `except:`, because that catches `KeyboardInterrupt`